codec/produce_req_test.go

```diff
@@ -83,3 +83,19 @@ func TestDecodeProduceReqV8(t *testing.T) {
 	assert.Equal(t, []byte("key"), record.Key)
 	assert.Equal(t, "value", string(record.Value))
 }
+
+func TestDecodeAndCodeReqV7(t *testing.T) {
```

Comment:
let's add a simple codec case. We may have arguments on codec function, like contains apiVersion or not.